arch: arm: stm32f2: remove core zephyr header inclusions from soc.h

ibisz opened the pull request:
The stm32f2 version of soc.h misses the changes done in commit
aee97be ("arch: arm: soc: remove core zephyr header inclusions
from soc.h").

Signed-off-by: Istvan Bisz <istvan.bisz@t-online.hu>

erwango commented on arch/arm/soc/st_stm32/stm32f2/soc.h:
```diff
@@ -23,12 +23,10 @@
-
 #include <stm32f2xx.h>
 
+#include <kernel_includes.h>
```
Can you add the matching comment as well?
```
/* ARM CMSIS definitions must be included before kernel_includes.h.
 * Therefore, it is essential to include kernel_includes.h after including
 * core SOC-specific headers.
 */
```
Thanks

ibisz committed:
arch: arm: stm32f2: remove core zephyr header inclusions from soc.h

The stm32f2 version of soc.h misses the changes done in commit
aee97be ("arch: arm: soc: remove core zephyr header inclusions
from soc.h").

Signed-off-by: Istvan Bisz <istvan.bisz@t-online.hu>

erwango reviewed (approved):
Thanks!
